frontend/model/notifications/mainNotificationsMixin.js

```diff
@@ -0,0 +1,193 @@
+    },
+    emit ({ rootGetters }) {
+      sbp('gi.notifications/emit', 'INCOME_DETAILS_OLD', {
+        createdDate: new Date().toISOString(),
+        months: 6,
+        lastUpdatedDate: rootGetters.ourGroupProfile.incomeDetailsLastUpdatedDate
+      })
+    }
+  }
+]
+
+const periodicNotificationEntries = [
+  {
+    type: PERIODIC_NOTIFICATION_TYPE.MIN1,
+    notificationData: {
+      stateKey: 'nearDistributionEnd',
+      emitCondition ({ rootGetters }) {
+        const currentPeriod = rootGetters.groupSettings?.distributionDate
+        if (!currentPeriod) { return false }
+
+        const nextPeriod = rootGetters.periodAfterPeriod(currentPeriod)
+        const now = new Date().toISOString()
+        const comparison = comparePeriodStamps(nextPeriod, now)
+
+        return rootGetters.ourGroupProfile?.incomeDetailsType === 'pledgeAmount' &&
+          (comparison > 0 && comparison < DAYS_MILLIS * 7) &&
+          (rootGetters.ourPayments && rootGetters.ourPayments.todo.length > 0) &&
+          !myNotificationHas(item => item.type === 'NEAR_DISTRIBUTION_END' && item.data.period === currentPeriod)
+      },
+      emit ({ rootState, rootGetters }) {
+        sbp('gi.notifications/emit', 'NEAR_DISTRIBUTION_END', {
+          createdDate: new Date().toISOString(),
+          groupID: rootState.currentGroupId,
+          period: rootGetters.groupSettings.distributionDate
+        })
+      },
+      shouldClearStateKey ({ rootGetters }) {
+        const currentPeriod = rootGetters.groupSettings.distributionDate
+        return rootGetters.currentNotifications.filter(item => item.type === 'NEAR_DISTRIBUTION_END')
+          .every(item => item.data.period !== currentPeriod)
+      }
+    }
+  },
+  {
+    type: PERIODIC_NOTIFICATION_TYPE.MIN1,
+    notificationData: {
+      stateKey: 'nextDistributionPeriod',
+      emitCondition ({ rootGetters }) {
+        if (!rootGetters.ourGroupProfile?.incomeDetailsType) return false // if income-details are not set yet, ignore.
+
+        const currentPeriod = rootGetters.groupSettings?.distributionDate
+        return currentPeriod &&
+          comparePeriodStamps(dateToPeriodStamp(new Date()), currentPeriod) > 0 &&
+          !myNotificationHas(item => item.type === 'NEW_DISTRIBUTION_PERIOD' && item.data.period === currentPeriod)
+      },
+      emit ({ rootState, rootGetters }) {
+        sbp('gi.notifications/emit', 'NEW_DISTRIBUTION_PERIOD', {
+          createdDate: new Date().toISOString(),
+          groupID: rootState.currentGroupId,
+          creator: rootGetters.ourUsername,
+          memberType: rootGetters.ourGroupProfile.incomeDetailsType === 'pledgeAmount' ? 'pledger' : 'receiver'
+        })
+      },
+      shouldClearStateKey ({ rootGetters }) {
+        return comparePeriodStamps(new Date().toISOString(), rootGetters.groupSettings.distributionDate) > 0
```

Comment:
`dateToPeriodStamp(new Date())` here too (and anywhere else I might've missed, please double-check, this is to prevent possibility of future bugs)